test/CMakeLists.txt

```diff
@@ -278,6 +296,20 @@ add_perf_test(test_conv_embed_db
 )
 endif()
 
+add_custom_test(test_conv_for_implicit_gemm_mlir SKIP_UNLESS_ALL
+COMMAND ${IMPLICITGEMM_MLIR_FWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 1024	14 14	--weights	1024 1024 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
+COMMAND ${IMPLICITGEMM_MLIR_FWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 256 56 56 --weights	256 256 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
+COMMAND ${IMPLICITGEMM_MLIR_FWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 128 58 58 --weights	128 128 3 3 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-backward-weights
+
+COMMAND ${IMPLICITGEMM_MLIR_BWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 1024	14 14	--weights	1024 1024 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-forward --disable-backward-weights
+COMMAND ${IMPLICITGEMM_MLIR_BWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 256 56 56 --weights	256 256 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-forward --disable-backward-weights
+COMMAND ${IMPLICITGEMM_MLIR_BWD_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 128 58 58 --weights	128 128 3 3 --pads_strides_dilations 0 0 1 1 1 1 --disable-forward --disable-backward-weights
+
+COMMAND ${IMPLICITGEMM_MLIR_WRW_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 1024	14 14	--weights	1024 1024 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-forward
+COMMAND ${IMPLICITGEMM_MLIR_WRW_ENVS} $<TARGET_FILE:test_conv2d> --verbose --input	64 256 56 56 --weights	256 256 1 1 --pads_strides_dilations 0 0 1 1 1 1 --disable-backward-data --disable-forward
```

Comment:
can we add a few non-NCHW configs to test ?

Reply:
Will do, but in a different PR. This is blocked by the implementation of validation code (from China team, I believe). Once validation of NHWC is ready, I will push a new PR that specifically enable/test for that layout.

Reply:
We don't have any kernels that support NHWC ATM. We need to write the specific test as well.

Reply:
Yet, please move in smallest possible steps.